### PR TITLE
Add Latex Rendering with remark-math and Katex Plugins

### LINKED ISSUE #333: Fix positioning, padding, sizing of images
When rendering latex images in a message, the position, padding, and size could be improved:

Screenshot 2024-01-13 at 11 30 57 AM

Here's an example chat:

https://chatcraft.org/c/humphd/HWrYbl4QNB52j3RNkyT4k

This likely happens for all images vs. the ones I'm doing here.

### PR DESCRIPTION
**Pull request by mingming-ma**
Fix #333 #335 

Need to add "formula in LaTeX by enclosing it with $$ display style" or similar to mention the dollar symbol in the prompt.

[Examples](https://mingming-equation.console-overthinker-dev.pages.dev/c/chatcraft_dev/y-R7-655HWbsBgYYMpEBj)

### CODE REVIEW

**Review by humphd — CHANGES_REQUESTED** [review]
We better update the default system prompt to include a line about using Latex/Katex for any math, so the model does it and we can render it.

File: src/lib/system-prompt.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ I follow these rules when responding:
 - Format ALL lines of code to 80 characters or fewer
 - Use Mermaid diagrams when discussing visual topics
 - If using functions, only use the specific functions I have been provided with
+- formula in LaTeX by enclosing it with $$ display style
```
Comment (humphd) [inline src/lib/system-prompt.ts:13]:
```suggestion
- If responding with math, use KaTeX syntax in your Markdown by enclosing it in $$...$$
```

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/lib/system-prompt.ts (hunk 1 above)
Comment (mingming-ma) [reply src/lib/system-prompt.ts]: image Seems a little bit confusing after rendering, let me try another approach

**Comment by tarasglek** [comment]
nice

File: src/lib/system-prompt.ts (hunk 1 above)
Comment (humphd) [reply src/lib/system-prompt.ts]: You could try these:

- If responding with math, use KaTeX syntax in your Markdown
- If responding with math, use KaTeX syntax in your Markdown by enclosing it in double-dollar signs

File: src/lib/system-prompt.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ I follow these rules when responding:
 - Format ALL lines of code to 80 characters or fewer
 - Use Mermaid diagrams when discussing visual topics
 - If using functions, only use the specific functions I have been provided with
-- If responding with math, use KaTeX syntax in your Markdown by enclosing it in $$...$$
+- If responding with math, use KaTeX syntax in your Markdown by enclosing it with the dollar symbol rather than parentheses, for example: $$O(nlogn)$$
```
Comment (humphd) [inline src/lib/system-prompt.ts:13]: '...with double-dollar signs...'

I like that you included an example, smart.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Comment by tarasglek** [comment]
This has a funny conflict with coding :(
image

image

**Comment by mingming-ma** [comment]
👍 somewhere need to configure so only accept $$, let me fix that

**Comment by humphd** [comment]
Or never render katex/math in code/pre blocks.

**Pull request merged**